packages/sit-onyx/src/components/OnyxRadioButton/OnyxRadioButton.vue

```diff
@@ -8,34 +9,128 @@ export type RadioButtonProps<TValue> = SelectionProps<TValue> & {
    * See also: https://html.spec.whatwg.org/multipage/input.html#radio-button-group
    */
   name: string;
+  isRequired?: boolean;
+  errorMessage?: string;
 };
 
 const props = defineProps<RadioButtonProps<TValue>>();
+
+const selector = ref<HTMLInputElement>();
+
+watch(
+  [() => props.errorMessage, selector],
+  () => selector.value?.setCustomValidity(props.errorMessage ?? ""),
+  { immediate: true },
+);
 </script>
 
 <template>
   <!-- TODO: decide on support prefix and/or folder -->
-  <label class="onyx-radio-button">
+  <label class="onyx-radio-button" :title="props.errorMessage">
     <!-- TODO: readonly is not supported on native radio input: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-readonly#description -->
+    <!-- TODO: accessible error: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-errormessage -->
     <input
+      ref="selector"
       class="onyx-radio-button__selector"
       type="radio"
+      :required="props.isRequired"
       :name="props.name"
       :value="props.id"
       :checked="props.selected"
-      :disabled="props.isDisabled"
+      :disabled="props.isDisabled || props.isReadonly"
     />
     <span class="onyx-radio-button__label">{{ props.label }}</span>
   </label>
 </template>
 
 <style lang="scss">
 .onyx-radio-button {
+  display: inline-flex;
+  align-items: center;
+  height: 2.5rem;
```

Comment:
Shouldn't the height be determined by the padding?

Reply:
Lets put this discussion off until we have a density concept